Observer/AddToCartInsuranceObserver.php

```diff
@@ -0,0 +1,180 @@
+    }
+
+	/**
+	 * @param Observer $observer
+	 * @return void
+	 */
+    public function execute(Observer $observer): void
+    {
+        try {
+            $insuranceProduct = $this->insuranceHelper->getAlmaInsuranceProduct();
+        } catch (AlmaInsuranceProductException $e) {
+            return;
+        }
+
+        /** @var Item $addedItemToQuote */
+        $addedItemToQuote = $observer->getData('quote_item');
+
+        if ($addedItemToQuote->getProduct()->getId() === $insuranceProduct->getId()) {
+            $this->logger->info('Warning insurance product added to quote', []);
+            return;
+        }
+
+        $insuranceId = $this->request->getParam('alma_insurance_id');
+        $insuranceQty = $this->request->getParam('alma_insurance_qty')??1;
+
+        if (!$insuranceId) {
+            $this->logger->info('Warning no insurance contract id in request params', [$this->request->getParams()]);
+            return;
+        }
+
+        $insuranceObject = $this->insuranceHelper->getInsuranceProduct(
+            $addedItemToQuote->getPrice(),
+            $this->configurableItemProductResolver->getFinalProduct($addedItemToQuote),
+            $insuranceId,
+            $addedItemToQuote->getQuoteId()
+        );
+        if (!$insuranceObject) {
+            $this->logger->info('Warning no insurance found for this contract id', [
+                'itemSku' => $addedItemToQuote->getSku(),
+                'insuranceId' => $insuranceId,
+                'itemPrice' => $addedItemToQuote->getPrice(),
+                'quoteId' => $addedItemToQuote->getQuoteId()
+            ]);
+            return;
+        }
+        $insuranceObject->setLinkToken($this->insuranceHelper->createLinkToken($addedItemToQuote->getProduct()->getId(), $insuranceObject->getId()));
+        try {
+            $this->insuranceHelper->setAlmaInsuranceToQuoteItem($addedItemToQuote, $insuranceObject->toArray(), InsuranceHelper::ALMA_PRODUCT_WITH_INSURANCE_TYPE);
+            $insuranceProductInQuote = $this->addInsuranceProductToQuote($addedItemToQuote->getQuote(), $insuranceProduct, $insuranceQty, $insuranceObject);
+            $this->insuranceHelper->setAlmaInsuranceToQuoteItem($insuranceProductInQuote, $insuranceObject->toArray(), InsuranceHelper::ALMA_INSURANCE_SKU);
+        } catch (\Exception $e) {
+            $this->logger->info('Error', [$e->getMessage()]);
+        }
+    }
+
+    /**
+     * @param Quote $quote
+     * @param Product $magentoInsuranceProduct
+     * @param int $qtyToAdd
+     * @param InsuranceProduct $insuranceProduct
+     * @return Item
+     * @throws AlmaInsuranceProductException
+     */
+    private function addInsuranceProductToQuote(Quote $quote, Product $magentoInsuranceProduct, int $qtyToAdd, InsuranceProduct $insuranceProduct): Item
+    {
+        try {
+            $insuranceInQuote = $quote->addProduct($magentoInsuranceProduct, $this->makeAddRequest($magentoInsuranceProduct, $qtyToAdd));
```

Comment:
Need explanation on this

Reply:
i edit Item in quote no Porduct in catalog